server/jetstream_test.go

```diff
@@ -16276,6 +16276,73 @@ func TestJetStreamConsumerAckSamplingSpecifiedUsingUpdateConsumer(t *testing.T)
 	}
 }
 
+func TestJetStreamConsumerMaxDeliveryUpdateConsumer(t *testing.T) {
```

Comment:
I would like to see also a test with cluster mode, to see if your changes handle that properly.

Reply:
Cluster mode tests added